alf/nest/nest.py

```diff
@@ -0,0 +1,329 @@
+# Copyright (c) 2020 Horizon Robotics. All Rights Reserved.
```

Comment:
add `alf/nest/__init__.py` and import the public functions of nest.py there so that they can be used as alf.nest.map_structure.